### ISSUE TITLE
Users in group edit window are listed multiple times

### ISSUE THREAD

**Issue by joberthel** [passbolt/passbolt_api#249]
## Users in group edit window are listed multiple times
* Passbolt Version: 2.0.4-debian
* Platform and Target:
-- Database server: mysql:5.7
-- docker
-- updated installation from v1

### What you did
I pressed the button "Edit group" in the groups tooltip.

### What happened
Sometimes all users in the list are listed multiple times. But without role information.

<img width="476" alt="bildschirmfoto 2018-04-30 um 09 58 07" src="https://user-images.githubusercontent.com/34104366/39418948-96f22ca2-4c5d-11e8-8b62-71cbb7d3062a.png">
<img width="479" alt="bildschirmfoto 2018-04-30 um 09 58 20" src="https://user-images.githubusercontent.com/34104366/39418949-99172bcc-4c5d-11e8-84fc-fac2937d259c.png">

### What you expected to happen
A user should only be listed once.

**Comment by stripthis** [comment]
HI @joberthel,

Could you clear your browser cache and try again? This could be due to the application javascript from v1 being used on v2.

**Comment by joberthel** [comment]
@stripthis 

Hi, I cleared the passbolt cache, which fixed the issue for me. Issue could have appeard because I deleted a user who was a member of the group. Will investigate further.

**Comment by stripthis** [comment]
You can check the database integrity using 
```
./bin/cake passbolt cleanup --dry-run
```

**Comment by joberthel** [comment]
It outputs: "No issue found, data looks squeaky clean!"

**Comment by stripthis** [comment]
So you should be fine :)

**Comment by joberthel** [comment]
The bug occurred again. Refreshing the page helped this time.

**Comment by cedricalfonsi** [comment]
Could you check the browser console (and network tab) and send us what you see.

**Comment by DanielBoettner** [comment]
I can second this issue.

I started a fresh docker container with 2.0.7 using a Database that already existed from a previous installation. Migration, healthcheck and cleanup had no errors.

**Comment by stripthis** [comment]
@DanielBoettner  could you check if you have the same collation / charset for all your tables?

**Comment by DanielBoettner** [comment]
Several things seem a bit off:

Thats from **show table status** for users_resources_permissions ` | NULL   |    NULL | NULL       | NULL |           NULL |        NULL |            NULL |         NULL |      NULL |           NULL | NULL                | NULL                | NULL       | NULL               |     NULL | NULL           | VIEW`

I can't tell if this intentional. As it is the only table without the InnoDB Engine.

The check for table collation gives this:

```SELECT TABLE_CATALOG, TABLE_SCHEMA, TABLE_NAME, COLUMN_NAME, COLLATION_NAME FROM INFORMATION_SCHEMA.COLUMNS`

returns 

```
+---------------+--------------------+---------------------------------------+-----------------------------------+--------------------+
| TABLE_CATALOG | TABLE_SCHEMA       | TABLE_NAME                            | COLUMN_NAME                       | COLLATION_NAME     |
+---------------+--------------------+---------------------------------------+-----------------------------------+--------------------+
| def           | passbolt           | authentication_tokens                 | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | authentication_tokens                 | token                             | utf8mb4_unicode_ci |
| def           | passbolt           | authentication_tokens                 | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | authentication_tokens                 | active                            | NULL               |
| def           | passbolt           | authentication_tokens                 | created                           | NULL               |
| def           | passbolt           | authentication_tokens                 | modified                          | NULL               |
| def           | passbolt           | burzum_file_storage_phinxlog          | version                           | NULL               |
| def           | passbolt           | burzum_file_storage_phinxlog          | migration_name                    | utf8_general_ci    |
| def           | passbolt           | burzum_file_storage_phinxlog          | start_time                        | NULL               |
| def           | passbolt           | burzum_file_storage_phinxlog          | end_time                          | NULL               |
| def           | passbolt           | burzum_file_storage_phinxlog          | breakpoint                        | NULL               |
| def           | passbolt           | comments                              | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | parent_id                         | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | foreign_key                       | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | foreign_model                     | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | content                           | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | created                           | NULL               |
| def           | passbolt           | comments                              | modified                          | NULL               |
| def           | passbolt           | comments                              | created_by                        | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | modified_by                       | utf8mb4_unicode_ci |
| def           | passbolt           | comments                              | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | id                                | NULL               |
| def           | passbolt           | email_queue                           | email                             | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | from_name                         | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | from_email                        | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | subject                           | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | config                            | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | template                          | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | layout                            | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | theme                             | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | format                            | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | template_vars                     | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | headers                           | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | sent                              | NULL               |
| def           | passbolt           | email_queue                           | locked                            | NULL               |
| def           | passbolt           | email_queue                           | send_tries                        | NULL               |
| def           | passbolt           | email_queue                           | send_at                           | NULL               |
| def           | passbolt           | email_queue                           | created                           | NULL               |
| def           | passbolt           | email_queue                           | modified                          | NULL               |
| def           | passbolt           | email_queue                           | attachments                       | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue_phinxlog                  | version                           | NULL               |
| def           | passbolt           | email_queue_phinxlog                  | migration_name                    | utf8_general_ci    |
| def           | passbolt           | email_queue_phinxlog                  | start_time                        | NULL               |
| def           | passbolt           | email_queue_phinxlog                  | end_time                          | NULL               |
| def           | passbolt           | email_queue_phinxlog                  | breakpoint                        | NULL               |
| def           | passbolt           | favorites                             | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | favorites                             | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | favorites                             | foreign_key                       | utf8mb4_unicode_ci |
| def           | passbolt           | favorites                             | foreign_model                     | utf8mb4_unicode_ci |
| def           | passbolt           | favorites                             | created                           | NULL               |
| def           | passbolt           | favorites                             | modified                          | NULL               |
| def           | passbolt           | file_storage                          | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | foreign_key                       | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | model                             | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | filename                          | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | filesize                          | NULL               |
| def           | passbolt           | file_storage                          | mime_type                         | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | extension                         | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | hash                              | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | path                              | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | adapter                           | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | created                           | NULL               |
| def           | passbolt           | file_storage                          | modified                          | NULL               |
| def           | passbolt           | gpgkeys                               | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | armored_key                       | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | bits                              | NULL               |
| def           | passbolt           | gpgkeys                               | uid                               | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | key_id                            | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | fingerprint                       | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | type                              | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | expires                           | NULL               |
| def           | passbolt           | gpgkeys                               | key_created                       | NULL               |
| def           | passbolt           | gpgkeys                               | deleted                           | NULL               |
| def           | passbolt           | gpgkeys                               | created                           | NULL               |
| def           | passbolt           | gpgkeys                               | modified                          | NULL               |
| def           | passbolt           | groups                                | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | groups                                | name                              | utf8mb4_unicode_ci |
| def           | passbolt           | groups                                | deleted                           | NULL               |
| def           | passbolt           | groups                                | created                           | NULL               |
| def           | passbolt           | groups                                | modified                          | NULL               |
| def           | passbolt           | groups                                | created_by                        | utf8mb4_unicode_ci |
| def           | passbolt           | groups                                | modified_by                       | utf8mb4_unicode_ci |
| def           | passbolt           | groups_users                          | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | groups_users                          | group_id                          | utf8mb4_unicode_ci |
| def           | passbolt           | groups_users                          | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | groups_users                          | is_admin                          | NULL               |
| def           | passbolt           | groups_users                          | created                           | NULL               |
| def           | passbolt           | permissions                           | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | aco                               | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | aco_foreign_key                   | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | aro                               | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | aro_foreign_key                   | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | type                              | NULL               |
| def           | passbolt           | permissions                           | created                           | NULL               |
| def           | passbolt           | permissions                           | modified                          | NULL               |
| def           | passbolt           | phinxlog                              | version                           | NULL               |
| def           | passbolt           | phinxlog                              | migration_name                    | utf8_general_ci    |
| def           | passbolt           | phinxlog                              | start_time                        | NULL               |
| def           | passbolt           | phinxlog                              | end_time                          | NULL               |
| def           | passbolt           | phinxlog                              | breakpoint                        | NULL               |
| def           | passbolt           | profiles                              | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | profiles                              | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | profiles                              | first_name                        | utf8mb4_unicode_ci |
| def           | passbolt           | profiles                              | last_name                         | utf8mb4_unicode_ci |
| def           | passbolt           | profiles                              | created                           | NULL               |
| def           | passbolt           | profiles                              | modified                          | NULL               |
| def           | passbolt           | resources                             | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | name                              | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | username                          | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | uri                               | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | description                       | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | deleted                           | NULL               |
| def           | passbolt           | resources                             | created                           | NULL               |
| def           | passbolt           | resources                             | modified                          | NULL               |
| def           | passbolt           | resources                             | created_by                        | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | modified_by                       | utf8mb4_unicode_ci |
| def           | passbolt           | roles                                 | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | roles                                 | name                              | utf8mb4_unicode_ci |
| def           | passbolt           | roles                                 | description                       | utf8mb4_unicode_ci |
| def           | passbolt           | roles                                 | created                           | NULL               |
| def           | passbolt           | roles                                 | modified                          | NULL               |
| def           | passbolt           | secrets                               | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | secrets                               | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | secrets                               | resource_id                       | utf8mb4_unicode_ci |
| def           | passbolt           | secrets                               | data                              | utf8mb4_unicode_ci |
| def           | passbolt           | secrets                               | created                           | NULL               |
| def           | passbolt           | secrets                               | modified                          | NULL               |
| def           | passbolt           | user_agents                           | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | user_agents                           | name                              | utf8mb4_unicode_ci |
| def           | passbolt           | users                                 | id                                | utf8mb4_unicode_ci |
| def           | passbolt           | users                                 | role_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | users                                 | username                          | utf8mb4_unicode_ci |
| def           | passbolt           | users                                 | active                            | NULL               |
| def           | passbolt           | users                                 | deleted                           | NULL               |
| def           | passbolt           | users                                 | created                           | NULL               |
| def           | passbolt           | users                                 | modified                          | NULL               |
| def           | passbolt           | users_resources_permissions           | user_id                           | utf8mb4_unicode_ci |
| def           | passbolt           | users_resources_permissions           | resource_id                       | utf8mb4_unicode_ci |
| def           | passbolt           | users_resources_permissions           | permission_id                     | utf8mb4_unicode_ci |
| def           | passbolt           | users_resources_permissions           | permission_type                   | NULL               |
+---------------+--------------------+---------------------------------------+-----------------------------------+--------------------+

```

and 

`SELECT TABLE_CATALOG, TABLE_SCHEMA, TABLE_NAME, TABLE_COLLATION
FROM INFORMATION_SCHEMA.TABLES;`

returns
```
+---------------+--------------------+---------------------------------------+--------------------+
| TABLE_CATALOG | TABLE_SCHEMA       | TABLE_NAME                            | TABLE_COLLATION    |
+---------------+--------------------+---------------------------------------+--------------------+
| def           | passbolt           | authentication_tokens                 | utf8mb4_unicode_ci |
| def           | passbolt           | burzum_file_storage_phinxlog          | utf8_general_ci    |
| def           | passbolt           | comments                              | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue                           | utf8mb4_unicode_ci |
| def           | passbolt           | email_queue_phinxlog                  | utf8_general_ci    |
| def           | passbolt           | favorites                             | utf8mb4_unicode_ci |
| def           | passbolt           | file_storage                          | utf8mb4_unicode_ci |
| def           | passbolt           | gpgkeys                               | utf8mb4_unicode_ci |
| def           | passbolt           | groups                                | utf8mb4_unicode_ci |
| def           | passbolt           | groups_users                          | utf8mb4_unicode_ci |
| def           | passbolt           | permissions                           | utf8mb4_unicode_ci |
| def           | passbolt           | phinxlog                              | utf8_general_ci    |
| def           | passbolt           | profiles                              | utf8mb4_unicode_ci |
| def           | passbolt           | resources                             | utf8mb4_unicode_ci |
| def           | passbolt           | roles                                 | utf8mb4_unicode_ci |
| def           | passbolt           | secrets                               | utf8mb4_unicode_ci |
| def           | passbolt           | user_agents                           | utf8mb4_unicode_ci |
| def           | passbolt           | users                                 | utf8mb4_unicode_ci |
| def           | passbolt           | users_resources_permissions           | NULL               |
+---------------+--------------------+---------------------------------------+--------------------+

```

I hope this helps.

**Comment by stripthis** [comment]
@DanielBoettner users_resources_permissions is a view from v1 that can be dropped. We attempt to drop it during migration, but there might have been some authorization issue. Apart from that your schema looks fine so the issue must be coming from something else. I have a "wild guess fix", which we will try to rollout in next version.

**Comment by DanielBoettner** [comment]
@stripthis 
I dropped the view.

Can you tell me the correct place to report a bug related to "encrypting" when updating password or groups?

In short, I have an error when I try to add a user to a group that was existing before my last update.
I thought about creating a new group and changing the owner of the passwords to the new group.
But I don't know if this would really solve the problems.

` Uncaught (in promise) TypeError: Cannot read property 'Secret' of undefined ` when trying to add a user to a group, in which I was not previously a group_manager.

and a bad request when trying to add a user to a group I always was managing.

`vendors.min.js:82340 PUT https://mydomain.tld:4343/groups/5a1d50a9-24c0-4525-878a-0096ac11000c.json?api-version=v1 400 (Bad Request)`

Please let me know on where to put the issue. Thanks in advance

**Comment by stripthis** [comment]
@DanielBoettner  you can open a new github issue, ideally with a way for us to reproduce the issue.

**Comment by stripthis** [comment]
I expect the "Users in group edit window are listed multiple times" issue to be fixed with v2.0.9. extension released today. Feel free to create a new issue (this one is messy and includes several issues now) if you encounter the issue again.
